var/spack/repos/builtin/packages/siesta/package.py

```diff
@@ -49,49 +98,209 @@ def flag_handler(self, name, flags):
+                "--with-scalapack=%s" % spec["scalapack"].libs,
+                # need to specify MPIFC explicitly below, otherwise
+                # Intel's mpiifort is not found
+                "MPIFC=%s" % spec["mpi"].mpifc,
+            ]
+            if "+cray" in spec:
+                configure_args += ["--with-netcdf={0}/lib/libnetcdff.so".format(netcdff_prefix)]
+            else:
+                configure_args += [
+                    "--with-netcdf=%s" % (spec["netcdf-fortran"].libs + spec["netcdf-c"].libs)
+                ]
+
+            if self.spec.satisfies("%gcc"):
+                if "+cray" in spec:
+                    configure_args.append(
+                        "FCFLAGS=-ffree-line-length-0 -I{0}/include".format(netcdff_prefix)
+                    )
+                else:
+                    configure_args.append("FCFLAGS=-ffree-line-length-0")
+            for d in ["Obj", "Obj_trans"]:
+                with working_dir(d, create=True):
+                    sh("../Src/configure", *configure_args)
+                    if spec.satisfies("@:4.0%intel"):
+                        with open("arch.make", "a") as f:
+                            f.write("\natom.o: atom.F\n")
+                            f.write("\t$(FC) -c $(FFLAGS) -O1")
+                            f.write("$(INCFLAGS) $(FPPFLAGS) $<")
+                    sh("../Src/obj_setup.sh")
+
+        elif self.spec.satisfies("@:4.1.5"):
+            with working_dir("Obj", create=True):
                 sh("../Src/obj_setup.sh")
+                if spec.satisfies("@:4.1.5%gcc"):
+                    shutil.copy("./gfortran.make", "./arch.make")
+                    libs_arg = []
+                    fppflags_arg = []
+                    arch_make = FileFilter("./arch.make")
+                    arch_make.filter(
+                        "FFLAGS = .*",
+                        "FFLAGS = {0}".format(
+                            "-O2 -fPIC -ftree-vectorize -fallow-argument-mismatch"
+                        ),
+                    )
+
+                    if "+debug" in spec:
+                        arch_make.filter("FFLAGS_DEBUG=.*", "FFLAGS_DEBUG= -g -O1")
+
+                    if "^cray-libsci" in spec:
+                        libs_arg.append("-L{0}/lib -lsci_gnu".format(spec["cray-libsci"].prefix))
+
+                    with open("arch.make", "a") as f:
+                        if "+mpi" in spec:
+                            arch_make.filter("CC = .*", "CC = {0}".format(spec["mpi"].mpicc))
+                            arch_make.filter("FC = .*", "FC = {0}".format(spec["mpi"].mpifc))
+                            if "^cray-libsci" in spec:
+                                libs_arg.append("-lsci_gnu_mpi")
+                            f.write("MPI_INTERFACE = libmpi_f90.a\n")
+                            f.write("MPI_INCLUDE = .\n")
+                            fppflags_arg.append("-DMPI ")
```

Comment:
After building with `+mpi` once, a simple `spack install siesta@4.1.5` now apparently selected `+mpi` (or a virtual provides changed, I assume) but that changed the final `mpif90 -o siesta` to fail with many undefined references including some to `blacs_get_`. I saw that the `blacs`/`scalapack` libs weren't added to `LIBS`, and this fixed the final link with +mpi on Ubuntu ha22.04:
```suggestion
                            fppflags_arg.append("-DMPI ")
                            f.write("LIBS += " + spec["scalapack"].libs.ld_flags + "\n")
```

Reply:
I added this change to https://github.com/CINES-HPC/spack/pull/1 that you can merge into this PR by approving and merging https://github.com/CINES-HPC/spack/pull/1.